src/build/__init__.py

```diff
@@ -223,116 +187,77 @@ def _parse_build_system_table(pyproject_toml: Mapping[str, Any]) -> Dict[str, An
 
 
 class ProjectBuilder:
-    """
-    The PEP 517 consumer API.
-    """
+    """The PEP 517 consumer API."""
+
+    @overload
+    def __init__(self, srcdir: PathType, *, python_executable: str = ..., runner: RunnerType = ...) -> None:
+        ...
+
+    @overload
+    def __init__(self, srcdir: PathType, *, isolated_env: env.IsolatedEnv, runner: RunnerType = ...) -> None:
+        ...
 
     def __init__(
         self,
         srcdir: PathType,
+        *,
         python_executable: str = sys.executable,
-        scripts_dir: Optional[str] = None,
-        runner: RunnerType = pep517.wrappers.default_subprocess_runner,
+        isolated_env: Optional[env.IsolatedEnv] = None,
```

Comment:
I don't like this. IMO it has a bad API separation. I think constructing from an isolated environment should be done via a custom constructor, `from_isolated_env`.

Reply:
Is it bad because you consider the isolated env to be a 'higher-level' construct, compared to the other params? Just trying to understand.